src/Raven.Server/Documents/Sharding/Subscriptions/ShardedSubscriptionWorker.cs

```diff
@@ -108,17 +108,19 @@ protected override (bool ShouldTryToReconnect, ServerNode NodeRedirectTo) CheckI
             // always try to reconnect until the task is canceled or assertLastConnectionFailure will throw when 'MaxErroneousPeriod' will elapse
             try
             {
-                assertLastConnectionFailure.Invoke();
-                var r = base.CheckIfShouldReconnectWorker(ex, assertLastConnectionFailure, onUnexpectedSubscriptionError, throwOnRedirectNodeNotFound: false);
-                if (_closedDueNoDocsLeft)
-                    return (ShouldTryToReconnect: false, NodeRedirectTo: null);
-
                 if (_state.CancellationTokenSource.IsCancellationRequested)
                 {
                     _processingCts.Cancel();
                     return (ShouldTryToReconnect: false, NodeRedirectTo: null);
                 }
 
+                assertLastConnectionFailure.Invoke();
+
+                var r = base.CheckIfShouldReconnectWorker(ex, assertLastConnectionFailure: null, onUnexpectedSubscriptionError, throwOnRedirectNodeNotFound: false);
```

Comment:
we invoke `assertLastConnectionFailure` 1 line above so there is no need to pass it to base method